### PR TITLE
[risk=low][no ticket] Add workspacePath(), analysisTabPath(), and dataTabPath() UI functions

### PR DESCRIPTION
**Pull request by jmthibault79**
Centralize the construction of AoU UI Workspace URLs instead of constructing them as-needed.

Example: `workspacePath(ns, wsid)` instead of `/workspaces/${ns}/${wsid}`.

We have many occasions to construct URLs to the analysis tab and data tab (and sub-URLs) so there are convenience functions for these as well: analysisTabPath() and dataTabPath()

This is motivated by my work to rename "wsid", "workspaceId", etc. to "terraName" - it will be easier to do a big renaming after these are centralized.

Tested a few pages locally where I was less confident, but I didn't do that for all changes here.  More scrutiny of the Data Apps changes would be welcome.
 

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [x] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/pages/runtimes-list.tsx [hunk 1]
```diff
@@ -139,7 +140,7 @@ export const RuntimesList = fp.flow(
                   // called from, for example:
                   // https://github.com/DataBiosphere/terra-ui/blob/4333c7b94d6ce10a6fe079361e98c2b6cc71f83a/src/pages/Environments.js#L420
                   getLink: (_, { namespace, name }) =>
-                    `/workspaces/${namespace}/${stringToSlug(name)}/analysis`,
+                    analysisTabPath(namespace, stringToSlug(name)),
```
Comment (jmthibault79) [inline ui/src/app/pages/runtimes-list.tsx:143]: fixes a bug!  the url is not actually /analysis in Prod.  This function returns the right URL in all environments.

File: ui/src/app/components/copy-modal.tsx [hunk 2]
```diff
@@ -332,7 +326,8 @@ const CopyModal = withCdrVersions()(
     }
 
     renderActionButton() {
-      const resourceType = toDisplay(this.props.resourceType);
```
Comment (jmthibault79) [inline ui/src/app/components/copy-modal.tsx:335]: renamed this to avoid accidents

File: ui/src/app/components/copy-modal.tsx [hunk 3]
```diff
@@ -344,19 +339,21 @@ const CopyModal = withCdrVersions()(
...
           </Button>
         );
       } else if (this.state.requestState === RequestState.SUCCESS) {
         const { namespace, id } = this.state.destination;
         return (
           <Button
-            path={`/workspaces/${namespace}/${id}/${ResourceTypeHomeTabs.get(
-              this.props.resourceType
-            )}`}
+            path={
```
Comment (jmthibault79) [inline ui/src/app/components/copy-modal.tsx:349]: tested locally - the URL is correct for notebooks and non-notebooks

File: ui/src/app/components/resource-list.tsx [hunk 4]
```diff
@@ -72,9 +72,9 @@ const WorkspaceNavigation = (props: NavProps) => {
     resource,
     style,
   } = props;
-  const tab = isNotebook(resource) ? analysisTabName : 'data';
-  const url = `/workspaces/${namespace}/${id}/${tab}`;
-
+  const url = isNotebook(resource)
```
Comment (jmthibault79) [inline ui/src/app/components/resource-list.tsx:75]: tested locally

**Review by dolbeew — APPROVED** [review]
LGTM. Everything seems to be working when testing locally.

**Pull request merged**